### ISSUE TITLE
Switch between miniforge (Apple Silicon build) and miniconda (Rosetta 2)

### ISSUE THREAD

**Issue by fblamanna** [conda-forge/miniforge#165]
I am just wondering how to have and switch between:

1. miniforge3 with native support for apple silicon
2. miniconda with all packages installed via Rosetta 2 emulation

on a Mac. 
This is because some environment works well under M1, but for other packages (not yet supported) I have to install them via emulation on a classic miniconda environment.

Is there a method to install both (I guess so) and switch between then to update the terminal zsh shell?

**Comment by isuruf** [comment]
```bash
CONDA_SUBDIR=osx-64 conda create -n rosetta python   # create a new environment called rosetta with intel packages.
conda activate rosetta
python -c "import platform;print(platform.machine())"
conda env config vars set CONDA_SUBDIR=osx-64  # make sure that conda commands in this environment use intel packages
conda deactivate rosetta
conda activate rosetta
echo "CONDA_SUBDIR: $CONDA_SUBDIR"
```

**Comment by fblamanna** [comment]
Thanks it works, just some warnings aobut overwriting the system SUBDIR when switching between arm64 and osx-64 environments.

**Comment by hmaarrfk** [comment]
Would you like to add a note in the readme explaining and documenting this others?

**Comment by richard-moss** [comment]
Thanks for the Q and answers!

I just posted a related Q about whether it's possible to mix osx-64 and osx-arm64 wheels when using the M1 native miniforge3. #172.  This might be a different approach to @fblamanna problem.

**Comment by avocardio** [comment]
Sorry this might be a stupid question, is there any way to just have miniforge and have it run all the non-supported packages through rosetta? I currently use anaconda3, don't really want to be switching between that and miniforge... how would I tell the environments apart?

**Comment by hmaarrfk** [comment]
Are you trying to run one program (single address space) that is dynamically linked to both x86 code (through Rosetta) and aarch64 code (native)?

**Comment by simonalford42** [comment]
@isuruf 's answer does not work for me 😢 
Running `python -c "import platform;print(platform.machine())"` gives `arm64`

**Comment by ollie-bell** [comment]
> ```shell
> CONDA_SUBDIR=osx-64 conda create -n rosetta python   # create a new environment called rosetta with intel packages.
> conda activate rosetta
> python -c "import platform;print(platform.machine())"
> conda env config vars set CONDA_SUBDIR=osx-64  # make sure that conda commands in this environment use intel packages
> conda deactivate rosetta
> conda activate rosetta
> echo "CONDA_SUBDIR: $CONDA_SUBDIR"
> ```

So, just to be explicitly clear, is this supposed to work if you only have Miniforge for arm64 installed? i.e. you do not need a separate installation for Miniconda?

**Comment by BastianZim** [comment]
Yes your current setup should be sufficient. You don't need to have a second installation. This will be activated per environment.

**Comment by rmwise** [comment]
I've managed to create the environment but when I run `jupyter lab` the correct kernel doesn't appear.

I've 2 python installed one on M1 and another on rosetta2 because I've specific packages that don't support M1 architecture.

So, I'm running those commands on a Rosetta2 terminal and it can't find the correct kernel version
